app/controllers/application_controller.rb

```diff
@@ -81,7 +81,7 @@ def set_gettext_locale_db
   end
 
   def require_mail
-    if User.current && !User.current.hidden? && User.current.mail.blank?
+    if User.current && !User.current.hidden? && User.current.mail.blank? && User.current.mail_enabled
```

Comment:
IMHO, it makes more sense to first check if the user has mail_enabled, and only if that is true, then check if the email address is blank.

Reply:
Fixed.